### DISCUSSION TITLE
Security Vulnerabilities

### DISCUSSION THREAD

**Discussion by macmillernz** [tobychui/zoraxy#167] [General]
The docker image is full of high severity vulnerabilities. I like Zoraxy and where it's headed but it IS NOT safe for the internet.
Is there any plan to fix this?

**Comment by tobychui** [comment]
You mean the docker base image or the Zoraxy project itself?

**Comment by PassiveLemon** [comment]
According to Docker Scout, Zoraxy (and/or it's dependencies) introduced these CVEs:
[GHSA-m425-mq94-257g](https://scout.docker.com/v/GHSA-m425-mq94-257g?s=github&n=grpc&ns=google.golang.org&t=golang&vr=%3C1.56.3&utm_source=hub&utm_medium=ExternalLink)
[CVE-2023-44487](https://scout.docker.com/v/CVE-2023-44487?s=github&n=grpc&ns=google.golang.org&t=golang&vr=%3C1.56.3&utm_source=hub&utm_medium=ExternalLink)
[GMS-2023-3788](https://scout.docker.com/v/GMS-2023-3788?s=gitlab&n=grpc&ns=google.golang.org&t=golang&vr=%3C1.56.3&utm_source=hub&utm_medium=ExternalLink)
[CVE-2024-24786](https://scout.docker.com/v/CVE-2024-24786?s=github&n=protobuf&ns=google.golang.org&t=golang&vr=%3C1.33.0&utm_source=hub&utm_medium=ExternalLink)

As far as whats recorded for these, they introduce DOS attack vectors. While obviously not desirable, aren't exactly major vulnerability concerns.

There are some more introduced by the Alpine base image but those can be resolved by updating package versions in the image.

@macmillernz What did you use to discover these vulnerabilities?

**Reply by macmillernz** [reply]
I've used trivy and anchore which both show several critical and high vulnerabilities.
And most recently Snyk, where I scanned zoraxydocker/zoraxy and passivelemon/zoraxy-docker.

**Reply by PassiveLemon** [reply]
passivelemon/zoraxy-docker is deprecated. That's where I originally published before I became an official maintainer for the Docker image.

**Reply by macmillernz** [reply]
Good to know. The Unraid template still points to the passivelemon image fyi.

**Reply by AzAel76** [reply]
The unraid template was published by IBRACORP and hasn't been updated since last year. 
I had to modify it to get the 3.xx versions working, I've spoken to those guys before, maybe I can poke them to update the template.

**Comment by leandroebner** [comment]
I got flaged with one major CVE being [CVE-2025-68121](https://nvd.nist.gov/vuln/detail/CVE-2025-68121).

<img width="1359" height="356" alt="image" src="https://github.com/user-attachments/assets/9c7ea5eb-5985-4c61-87e9-9c35a9c84de7" />

I'm not sure if this is something to be concerned about, probably this is known already.
